### ISSUE TITLE
ML approach to predict changing epistasis from the polarity network

### ISSUE THREAD

**Issue by EKingma** [leilaicruz/machine-learning-for-yeast#8]
The idea is to make a specific case to predict how interactions change after the deletion of 1/2/multiple genes starting from the specific case of the polarity network, which would allow us to include data specific to this network in addition to the genome-wide data. 

A first idea of a dataset to use is the information of the **different combinations of polarity genes** in different yeast species: https://academic.oup.com/gbe/article/10/7/1765/5040802 

In addition to that, we could also use **phenotype data (cell morphology)** from here:
http://www.yeast.ib.k.u-tokyo.ac.jp/SCMD/index.php

**Comment by leilaicruz** [comment]
> The idea is to make a specific case to predict how interactions change after the deletion of 1/2/multiple genes starting from the specific case of the polarity network

- Here,  if a supervised algorithm is required you will need data from 1/2/multiple gene deletions with the corresponding output, let say known interactions.  You have to think how many cases of epistasis you know from cell polarity ? and those questions will guide you through the feasibility or one or another ML problem. 

> A first idea of a dataset to use is the information of the **different combinations of polarity genes** in different yeast species: https://academic.oup.com/gbe/article/10/7/1765/5040802

- Could you download the data and have a sense of what type of info we can extract from it? 

> In addition to that, we could also use **phenotype data (cell morphology)** from here:
> http://www.yeast.ib.k.u-tokyo.ac.jp/SCMD/index.php

- Also , in this case, can you play with the specific data you would like to use to see how can we combine both in a meaningful way? 

These are all questions that I had to make to myselft in order to properly apply a ML algorithm and to have control on what specifically I am giving as features in order to make meaningful interpretations afterwards.

**Comment by EKingma** [comment]
> > The idea is to make a specific case to predict how interactions change after the deletion of 1/2/multiple genes starting from the specific case of the polarity network
> 
> * Here,  if a supervised algorithm is required you will need data from 1/2/multiple gene deletions with the corresponding output, let say known interactions.  You have to think how many cases of epistasis you know from cell polarity ? and those questions will guide you through the feasibility or one or another ML problem.
> 
This is true, and this is also sort of the problem, since there is not a lot of data on that (other than the data we have from the bem1 deletion experiment and maybe data we will get from the SATAY). For now the plan is to indirectly include this in the form of which polarity protein combinations appear to exist and which ones do not (with the assumption that combinations that do not occur in nature lead to an inviable phenotype) 
> > A first idea of a dataset to use is the information of the **different combinations of polarity genes** in different yeast species: https://academic.oup.com/gbe/article/10/7/1765/5040802
> 
> * Could you download the data and have a sense of what type of info we can extract from it?
[evy121_supp.zip](https://github.com/leilaicruz/machine-learning-for-yeast/files/4717715/evy121_supp.zip)
The nice datafile seems to be the one with the 'protein repetoire', which contains the different protein combinations across different species (might need to be filtered for the right species). 
> 
> > In addition to that, we could also use **phenotype data (cell morphology)** from here:
> > http://www.yeast.ib.k.u-tokyo.ac.jp/SCMD/index.php
> 
> * Also , in this case, can you play with the specific data you would like to use to see how can we combine both in a meaningful way?
This database is very big, but I believe it includes bud-ratio/cell size, which should be a good measure for the effect of different gene deletions on polarity establishment. 
> 
> These are all questions that I had to make to myselft in order to properly apply a ML algorithm and to have control on what specifically I am giving as features in order to make meaningful interpretations afterwards.

**Comment by leilaicruz** [comment]
Ok , so go for it! and try to make first something simple, to grasp the workflow , and then try it more complex. Usually it is nice to draw (literally draw like in a piece of paper)  a matrix , to define which do you want to be columns which will be the features that describes each of your samples(each row). 
I suggest if you dont feel still comfortable in Python or R, you can do it in Matlab, at this point the language does not matter to much , it is just to learn to apply the ML workflow. 

Keep us informed about the progress!
